libsolidity/ast/Types.cpp

```diff
@@ -613,6 +613,9 @@ TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other
 		}
 		else if (dynamic_cast<FixedPointType const*>(_other))
 			return nullptr;
+		else if (auto rationalNumberType = dynamic_cast<RationalNumberType const*>(_other))
+			if (!rationalNumberType->integerType())
+				return TypeResult::err("Exponent too large.");
```

Comment:
Sorry to bother you again, but I think another reason for `integerType()` to return nullptr could be that the exponent is fractional: `x**(1/2)` - can you check, please?

Reply:
Good point. `integerType()` would throw if fractional. I'll add a check and an error before.